test/framework/robot.py

```diff
@@ -424,15 +424,15 @@ def test_resolve_dependencies_minimal(self):
             # to test resolving of dependencies with minimal toolchain
             # for each of these, we know test easyconfigs are available (which are required here)
             "dependencies = [",
-            "   ('OpenMPI', '2.1.2'),",  # available with GCC/6.4.0-2.28
+            # the use of %(version_minor)s here is mainly to check if templates are being handled correctly
+            # (it doesn't make much sense, but it serves the purpose)
+            "   ('OpenMPI', '%(version_minor)s.1.2'),",  # available with GCC/6.4.0-2.28
             "   ('OpenBLAS', '0.2.20'),",  # available with GCC/6.4.0-2.28
             "   ('ScaLAPACK', '2.0.2', '-OpenBLAS-0.2.20'),",  # available with gompi/2018a
             "   ('SQLite', '3.8.10.2'),",
             "]",
             # toolchain as list line, for easy modification later;
-            # the use of %(version_minor)s here is mainly to check if templates are being handled correctly
-            # (it doesn't make much sense, but it serves the purpose)
-            "toolchain = {'name': 'foss', 'version': '%(version_minor)s018a'}",
```

Comment:
This test was failing when this template was left in place, it wasn't clear to me where but my guess was that the new template triggers a version check of the toolchain somewhere which is now taking place before the templates have been resolved.

Reply:
See https://github.com/easybuilders/easybuild-framework/runs/545312074 for example

Reply:
Even after looking into it again, I still can't figure out *why* it happens, the error is:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/alanc/EasyBuild_Git/easybuild-framework/test/framework/robot.py", line 457, in test_resolve_dependencies_minimal
    res = resolve_dependencies([bar], self.modtool)
  File "/home/alanc/EasyBuild_Git/easybuild-framework/easybuild/tools/robot.py", line 460, in resolve_dependencies
    raise_error_missing_deps(totally_missing, extra_msg="no easyconfig file or existing module found")
  File "/home/alanc/EasyBuild_Git/easybuild-framework/easybuild/tools/robot.py", line 323, in raise_error_missing_deps
    raise EasyBuildError(error_msg)
EasyBuildError: 'Missing dependencies: foss/%(version_minor)s018a (no easyconfig file or existing module found)'
```
but the dict of the easyconfig is perfectly fine with templates resolved.

Reply:
@boegel I've spent an hour trying to figure out why but I cannot solve this one

Reply:
OK, here's what's going on...

The first time that the template dict is being composed (with the first call to `EasyConfig._generate_template_values`), `self.template_values` is still entirely blank.

But when `self.toolchain` is being passed to `template_constant_dict`, we're creating a toolchain instance, which requires the easyconfig file for the toolchain itself. So the toolchain version can't include template values like `%(version_minor)s` anymore now, since that template can't be resolved when `self.toolchain` is first used...

Strictly speaking that's a regression, but using `%(version_minor)s` to define the toolchain version is a pretty strange use case, so I don't consider this to be a problem.

Long story short: this test was doing something pretty strange, and the change you made to move it to `dependencies` makes sense.

This test was added in the context of a fix for the use of templates in `dependencies` (see #1812), so the changed test should still cover that use case.

Reply:
Actually, there's a clean fix for it, implemented in e763c3040...